src/web/tools/KeyboardEventManager.ts

```diff
@@ -0,0 +1,93 @@
+import { AdaptedEvent, EventTypes } from '../interfaces';
+import EventManager from './EventManager';
+import { PointerType } from '../../PointerType';
+
+export default class KeyboardEventManager extends EventManager<HTMLElement> {
+  private activationKeys = ['Enter', ' '];
+  private cancelationKeys = ['Tab'];
+  private isPressed = false;
+
+  private keyDownCallback = (event: KeyboardEvent): void => {
+    if (this.cancelationKeys.indexOf(event.key) !== -1 && this.isPressed) {
+      this.dispatchEvent(event, EventTypes.CANCEL);
+      return;
+    }
+
+    if (this.activationKeys.indexOf(event.key) === -1) {
+      return;
+    }
+
+    this.dispatchEvent(event, EventTypes.DOWN);
+  };
+
+  private keyUpCallback = (event: KeyboardEvent): void => {
+    if (this.activationKeys.indexOf(event.key) === -1 || !this.isPressed) {
+      return;
+    }
+
+    this.dispatchEvent(event, EventTypes.UP);
+  };
+
+  private dispatchEvent(event: KeyboardEvent, eventType: EventTypes) {
+    const adaptedEvent = this.mapEvent(event, eventType);
+
+    if (isNaN(adaptedEvent.x)) {
+      // in rare cases when event.target is not a HTMLElement
+      // we want to return before sending out an event
+      return;
+    }
```

Comment:
In that case, why not simply

```suggestion
    if (!(event.target instanceof HTMLElement)) {
      return;
    }
    const adaptedEvent = this.mapEvent(event, eventType);
```
?

Reply:
Great call, applied in [f72af6c](https://github.com/software-mansion/react-native-gesture-handler/pull/3035/commits/f72af6cd6a0917319c2cc381bc61f3d4d42090ad)